pkg/controller/machineset_util.go

```diff
@@ -75,3 +76,48 @@ func GetMachineSetHash(is *v1alpha1.MachineSet, uniquifier *int32) (string, erro
 	isTemplate.Labels = labelsutil.CloneAndRemoveLabel(isTemplate.Labels, v1alpha1.DefaultMachineDeploymentUniqueLabelKey)
 	return fmt.Sprintf("%d", ComputeHash(isTemplate, uniquifier)), nil
 }
+
+// syncMachinesNodeTemplates updates all machines in the given machineList with the new nodeTemplate if required.
+func (c *controller) syncMachinesNodeTemplates(machineList []*v1alpha1.Machine, machineSet *v1alpha1.MachineSet) error {
+
+	controlClient := c.controlMachineClient
+	machineLister := c.machineLister
+
+	for _, machine := range machineList {
+		// Ignore inactive Machines.
+		if !IsMachineActive(machine) {
+			continue
+		}
+
+		nodeTemplateChanged := copyMachineSetNodeTemplatesToMachines(machineSet, machine)
+		// Only sync the machine that doesn't already have the latest nodeTemplate.
+		if nodeTemplateChanged {
+			_, err := UpdateMachineWithRetries(controlClient.Machines(machine.Namespace), machineLister, machine.Namespace, machine.Name,
+				func(machineToUpdate *v1alpha1.Machine) error {
+					return nil
+				})
+			if err != nil {
+				return fmt.Errorf("error in updating nodeTemplateSpec to machine %q: %v", machine.Name, err)
+			}
+			glog.V(2).Infof("Updated machine %s/%s of MachineSet %s/%s with latest nodeTemplate.", machine.Namespace, machine.Name, machineSet.Namespace, machineSet.Name)
+		}
+	}
+	return nil
+}
+
+// copyMachineSetNodeTemplatesToMachines copies machineset's nodeTemplate to machine's nodeTemplate,
+// and returns true if machine's nodeTemplate is changed.
+// Note that apply and revision nodeTemplates are not copied.
+func copyMachineSetNodeTemplatesToMachines(machineset *v1alpha1.MachineSet, machine *v1alpha1.Machine) bool {
```

Comment:
Same comments as [`copyMachineDeploymentNodeTemplatesToMachineSet`](https://github.com/gardener/machine-controller-manager/pull/256/files#diff-3e28f59dda362e1974d0f3df011e28f1R434).

Reply:
Basically, the same answer, hope though I didn't miss anything in there.